docs/v3/core/index.rst

```diff
@@ -1599,46 +1471,266 @@ Storage transformers may be stacked to combine different functionalities:
+array       data type               :ref:`data-type <array-metadata-data-type>`                        :ref:`data-type-list`
+array       chunk grid              :ref:`chunk-grid <array-metadata-chunk-grid>`                      :ref:`chunk-grid-list`
+array       chunk key encoding      :ref:`chunk-key-encoding <array-metadata-chunk-key-encoding>`      :ref:`chunk-key-encoding-list`
+array       codecs                  :ref:`codecs <array-metadata-codecs>`                              :ref:`codec-list`
+array       storage transformer     :ref:`storage-transformers <array-metadata-storage-transformers>`  :ref:`storage-transformer-list`
+=========== ======================= ================================================================== ================================
+
+Note, that ``fill_value`` is not its own extension point, but is dependent on the data type.
+
+New extension points may be proposed to the Zarr community through the ZEP
+process. See `ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ for more information.
+
+.. _extension-definition:
+
+Extension definition
+--------------------
+
+.. _extension-definition-object:
+
+Objects
+^^^^^^^
+
+In `metadata documents`_, extensions can be encoded either as objects or as
+short-hand names.
+
+If using an object definition, the member ``name``
+MUST be a plain string which conforms to :ref:`extension name <extension-naming>`.
+Optionally, the member ``configuration`` MAY be present but if so MUST be
+an object.
+
+For example::
+
+    {
+        "name": "<name>",        # conformant name
+        "configuration": { ... } # optional object
+    }
+
+.. _extension-definition-short-hand-name:
+
+Short-hand names
+^^^^^^^^^^^^^^^^
+
+Instead of extension objects, short-hand names MAY be used if no
+configuration metadata is required. They are equivalent to extension
+objects with just a `name` key.
+
+.. _extension-definition-must-understand:
+
+`must_understand`
+^^^^^^^^^^^^^^^^^
+
+An extension object is interpreted to have an implicit field `must_understand` set to
+`True`, unless otherwise stated. An extension object MAY explicitly set `must_understand=False` if
+implementations can ignore its presence.
+
+An implementation MUST fail to open Zarr groups or arrays if any
+metadata fields are present which (a) the
+implementation does not recognize and (b) are not explicitly
+set to ``"must_understand": false``.
+
+`must_understand=False` is not supported for the following extension points:
+data type, chunk grid, and chunk key encoding.
+
+Use of `must_understand=False` to add top-level keys is discouraged in favor
+of the explicit use of :ref:`extension-points`.
+
+.. _extension-naming:
+
+Extension naming
+----------------
+
+The `name` field of an extension is an identifier taking one of two forms:
+**registered names** (as simple strings) and **unregistered names** (as URIs).
```

Comment:
i'm confused, going by [this comment](https://github.com/zarr-developers/zarr-specs/pull/330#issuecomment-2724936994) I thought we were dropping the requirement that unregistered names be URIs?

Reply:
so to be clear, is it required that unregistered names be URIs?

Reply:
Short answer: Yes. But, I hadn't realized that it wasn't generally accepted that URIs need not be `http://host/stuff`. So what I've tried to make clear in the text is that there are also [`urn:`](https://en.wikipedia.org/wiki/Uniform_Resource_Name)s and that we're going to pick a short and easy to understand  prefix for any unregistered/namespaced name. (The [`tag:` scheme](https://en.wikipedia.org/wiki/Tag_URI_scheme) may be of interest as well.)